libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs

```diff
@@ -338,7 +354,10 @@ private async Task<InputState> RecognizeInput(DialogContext dc)
             if (input == null)
             {
                 var turnCount = dc.State.GetValue<int>(TURN_COUNT_PROPERTY);
-                if (turnCount == 0)
+                var processInput = dc.State.GetValue<bool>(PROCESS_INPUT_PROPERTY, false);
+
+                // Go down this path only if the user has not requested to re-process user input via turn.processInput = true.
+                if (turnCount == 0 && !processInput)
```

Comment:
This looks odd to me. 

turnCount == 0 only designed to use in BeginDialog and only indicate that this is called the first time, so it will check dialog.value instead of turn.text (Because turn.text haven't even arrive)

so, i don't think we don't want processInput to be in this business, this processInput should designed as "reprocessInput". Which should work in ContinueDialog(as your very initial fix), not here.

Reply:
turnCount == 0 is even a bad design here, a more decoupled design is we make this function take a parameter called input, and passed different input according to context, so in BeginDialog will pick dialog.value and pass it here, in ContinueDialog it will pick turn.activity.text and pass it to here. You see?

Reply:
Here is an example that perhaps you don't want but made possible in this case

Steps:
   SetProperty("turn.processInput", true)
   NumberInput("what's you age")  \\ in here, your code will make this input to take the previous activity.text, i assume, which perhaps not you want? or is it?

Reply:
Confirmed with Vishwac, this is because we clear state after setProperty, so no change to fix that  in ContinueDialog.